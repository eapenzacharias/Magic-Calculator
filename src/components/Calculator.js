import React, { useState } from 'react';
import Display from './Display';
import RenderCol from './ButtonPanel';
import Button from './Button';
import calculate from '../logic/calculate';
import './style/style.scss';

const Calculator = () => {
  const [state, setState] = useState({
    total: '0',
    next: null,
    operations: null,
  });

  const handleClick = (buttonName) => {
    const cal = state;
    setState(calculate(cal, buttonName));
  };

  return (
    <div className="container main">
      <Display value={state.next || state.total || '0'} />
      <div className="button-panel">
        <RenderCol clickHandler={handleClick} btn0="AC" btn1="+/-" btn2="%" btn3="รท" />
        <RenderCol clickHandler={handleClick} btn0="7" btn1="8" btn2="9" btn3="x" />
        <RenderCol clickHandler={handleClick} btn0="4" btn1="5" btn2="6" btn3="-" />
        <RenderCol clickHandler={handleClick} btn0="1" btn1="2" btn2="3" btn3="+" />
        <div className="btn-col">
          <Button clickHandler={handleClick} buttonName="0" wide="wide" color="grey" />
          <Button clickHandler={handleClick} buttonName="." color="grey" />
          <Button clickHandler={handleClick} buttonName="=" />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
