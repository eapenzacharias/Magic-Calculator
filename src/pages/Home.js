import React from 'react';

const Home = () => {
  const homeData = {
    title: 'Math magicians',
    subHeading: 'Welcome to this page!',
    description:
      (
        <>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Pellentesque quam tellus, lobortis in neque id, consectetur mollis nisi.
            Integer dictum, augue eget volutpat cursus, tellus arcu semper libero,
            at eleifend risus magna sed neque. In hac habitasse platea dictumst.
            Vivamus est risus, auctor pulvinar egestas et, venenatis at lacus.
            Donec placerat enim luctus, feugiat mauris sit amet, dapibus mi.
            Aliquam a augue id urna porta tincidunt. Nunc dapibus, eros nec
            pellentesque tristique, nulla nunc malesuada enim, sed consequat quam
            lorem nec orci. Ut magna nisl, venenatis feugiat fringilla vel, mattis
            ac purus. Nam at sem eget augue scelerisque viverra vitae at metus.
            Vestibulum elit tortor, convallis sed turpis a, feugiat pellentesque urna.
            Etiam consectetur faucibus tincidunt. Nullam ornare faucibus nibh at sollicitudin.
            Aliquam tincidunt ligula in orci venenatis, at facilisis urna molestie.
          </p>
          <p>
            Donec placerat enim luctus, feugiat mauris sit amet, dapibus mi.
            Aliquam a augue id urna porta tincidunt. Nunc dapibus, eros nec
            pellentesque tristique, nulla nunc malesuada enim, sed consequat quam
            lorem nec orci. Ut magna nisl, venenatis feugiat fringilla vel, mattis
            ac purus. Nam at sem eget augue scelerisque viverra vitae at metus.
            Vestibulum elit tortor, convallis sed turpis a, feugiat pellentesque urna.
            Etiam consectetur faucibus tincidunt. Nullam ornare faucibus nibh at sollicitudin.
            Aliquam tincidunt ligula in orci venenatis, at facilisis urna molestie.
          </p>
        </>
      ),
  };

  const Page = () => {
    const { title, subHeading, description } = homeData;
    return (
      <>
        <div className="col">
          <h1>{title}</h1>
          <h2>{subHeading}</h2>
          <div>{description}</div>
        </div>
      </>
    );
  };
  return (
    <Page />
  );
};

export default Home;
