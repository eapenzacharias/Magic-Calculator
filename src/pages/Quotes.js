import React, { useEffect, useState } from 'react';

const Quotes = () => {
  const [activeSlide, setActiveSlide] = useState(1);

  useEffect(() => {
    // This variable prevents race condition
    let current = 1;
    const cycleReviews = () => {
      if (current === 5) {
        current = 1;
      } else {
        current += 1;
      }
      setActiveSlide(current);
    };
    // intervalId identified so it can be canceled on unmount
    const intervalId = setInterval(() => {
      cycleReviews();
    }, 10000);
    // Removes interval on unmount
    return () => clearInterval(intervalId);
  }, []);

  const reviews = [
    {
      name: 'Katherine Johnson',
      citation: 'African-American mathematician',
      quote:
        'We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics.',
    },
    {
      name: 'Richard Courant',
      citation: 'German-American mathematician',
      quote:
        'Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.',
    },
    {
      name: 'Marston Morse',
      citation: 'American mathematician',
      quote:
        'Mathematics are the result of mysterious powers which no one understands, and which the unconscious recognition of beauty must play an important part. Out of an infinity of designs a mathematician chooses one pattern for beauty’s sake and pulls it down to earth.',
    },
    {
      name: 'Shakuntala Devi',
      citation: 'Indian writer and mental calculator',
      quote:
        'Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers.',
    },
    {
      name: 'John von Neumann',
      citation: 'Hungarian mathematician',
      quote:
        'If people do not believe that mathematics is simple, it is only because they do not realize how complicated life is.',
    },
  ];

  return (
    <div className="carousel">
      <ul className="carousel__list">
        {reviews.map((review, index) => {
          const { citation, name, quote } = review;
          const count = index + 1;
          return (
            <li
              className={`carousel__item
                ${count === activeSlide ? ' active' : ''}
                ${count < activeSlide ? ' left' : ''}
                ${count > activeSlide ? ' right' : ''}
              `}
              key={count}
            >
              <blockquote className="carousel__quote">
                <cite>
                  <span className="carousel__name">{name}</span>
                  <span className="carousel__citation">{citation}</span>
                </cite>
                <p>
                  &quot;
                  {quote}
                  &quot;
                </p>
              </blockquote>
            </li>
          );
        })}
        <li className="carousel__indicator">
          <span
            className={`carousel__dot${activeSlide === 1 ? ' active' : ''}`}
          />
          <span
            className={`carousel__dot${activeSlide === 2 ? ' active' : ''}`}
          />
          <span
            className={`carousel__dot${activeSlide === 3 ? ' active' : ''}`}
          />
          <span
            className={`carousel__dot${activeSlide === 4 ? ' active' : ''}`}
          />
          <span
            className={`carousel__dot${activeSlide === 5 ? ' active' : ''}`}
          />
        </li>
      </ul>
    </div>
  );
};

export default Quotes;
