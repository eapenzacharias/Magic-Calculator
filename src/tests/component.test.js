import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import Calculator from '../components/Calculator';
import Quotes from '../pages/Quotes';
import Home from '../pages/Home';

describe('Component snapshots should match', () => {
  it('Nav bar should render', () => {
    const render = TestRenderer.create(<Router><Navbar /></Router>).toJSON();
    expect(render).toMatchSnapshot();
  });

  it('Homepage should render', () => {
    const render = TestRenderer.create(<Home />).toJSON();
    expect(render).toMatchSnapshot();
  });

  it('Calculator should render', () => {
    const render = TestRenderer.create(<Calculator />).toJSON();
    expect(render).toMatchSnapshot();
  });

  it('Quotes should render', () => {
    const render = TestRenderer.create(<Quotes />).toJSON();
    expect(render).toMatchSnapshot();
  });
});
