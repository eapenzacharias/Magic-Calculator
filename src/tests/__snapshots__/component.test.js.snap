// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshots should match Calculator should render 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="button-panel"
  >
    <div
      className="btn-col"
    >
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn small bg-orange"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="btn-col"
    >
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn small bg-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="btn-col"
    >
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn small bg-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="btn-col"
    >
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn small bg-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btn-col"
    >
      <button
        className="btn wide bg-grey"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn small bg-grey"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn small bg-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Component snapshots should match Homepage should render 1`] = `
<div
  className="col home"
>
  <h1>
    Math magicians
  </h1>
  <h2>
    Welcome to this page!
  </h2>
  <div>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quam tellus, lobortis in neque id, consectetur mollis nisi. Integer dictum, augue eget volutpat cursus, tellus arcu semper libero, at eleifend risus magna sed neque. In hac habitasse platea dictumst. Vivamus est risus, auctor pulvinar egestas et, venenatis at lacus. Donec placerat enim luctus, feugiat mauris sit amet, dapibus mi. Aliquam a augue id urna porta tincidunt. Nunc dapibus, eros nec pellentesque tristique, nulla nunc malesuada enim, sed consequat quam lorem nec orci. Ut magna nisl, venenatis feugiat fringilla vel, mattis ac purus. Nam at sem eget augue scelerisque viverra vitae at metus. Vestibulum elit tortor, convallis sed turpis a, feugiat pellentesque urna. Etiam consectetur faucibus tincidunt. Nullam ornare faucibus nibh at sollicitudin. Aliquam tincidunt ligula in orci venenatis, at facilisis urna molestie.
    </p>
    <p>
      Donec placerat enim luctus, feugiat mauris sit amet, dapibus mi. Aliquam a augue id urna porta tincidunt. Nunc dapibus, eros nec pellentesque tristique, nulla nunc malesuada enim, sed consequat quam lorem nec orci. Ut magna nisl, venenatis feugiat fringilla vel, mattis ac purus. Nam at sem eget augue scelerisque viverra vitae at metus. Vestibulum elit tortor, convallis sed turpis a, feugiat pellentesque urna. Etiam consectetur faucibus tincidunt. Nullam ornare faucibus nibh at sollicitudin. Aliquam tincidunt ligula in orci venenatis, at facilisis urna molestie.
    </p>
  </div>
</div>
`;

exports[`Component snapshots should match Nav bar should render 1`] = `
<nav
  className="navBar"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Component snapshots should match Quotes should render 1`] = `
<div
  className="row"
>
  <div
    className="carousel"
  >
    <ul
      className="carousel__list"
    >
      <li
        className="carousel__item
                 active
                
                
              "
      >
        <blockquote
          className="carousel__quote"
        >
          <cite>
            <span
              className="carousel__name"
            >
              Katherine Johnson
            </span>
            <span
              className="carousel__citation"
            >
              African-American mathematician
            </span>
          </cite>
          <p>
            "
            We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering, and technology. And there will always, always be mathematics.
            "
          </p>
        </blockquote>
      </li>
      <li
        className="carousel__item
                
                
                 right
              "
      >
        <blockquote
          className="carousel__quote"
        >
          <cite>
            <span
              className="carousel__name"
            >
              Richard Courant
            </span>
            <span
              className="carousel__citation"
            >
              German-American mathematician
            </span>
          </cite>
          <p>
            "
            Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
            "
          </p>
        </blockquote>
      </li>
      <li
        className="carousel__item
                
                
                 right
              "
      >
        <blockquote
          className="carousel__quote"
        >
          <cite>
            <span
              className="carousel__name"
            >
              Marston Morse
            </span>
            <span
              className="carousel__citation"
            >
              American mathematician
            </span>
          </cite>
          <p>
            "
            Mathematics are the result of mysterious powers which no one understands, and which the unconscious recognition of beauty must play an important part. Out of an infinity of designs a mathematician chooses one pattern for beauty’s sake and pulls it down to earth.
            "
          </p>
        </blockquote>
      </li>
      <li
        className="carousel__item
                
                
                 right
              "
      >
        <blockquote
          className="carousel__quote"
        >
          <cite>
            <span
              className="carousel__name"
            >
              Shakuntala Devi
            </span>
            <span
              className="carousel__citation"
            >
              Indian writer and mental calculator
            </span>
          </cite>
          <p>
            "
            Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers.
            "
          </p>
        </blockquote>
      </li>
      <li
        className="carousel__item
                
                
                 right
              "
      >
        <blockquote
          className="carousel__quote"
        >
          <cite>
            <span
              className="carousel__name"
            >
              John von Neumann
            </span>
            <span
              className="carousel__citation"
            >
              Hungarian mathematician
            </span>
          </cite>
          <p>
            "
            If people do not believe that mathematics is simple, it is only because they do not realize how complicated life is.
            "
          </p>
        </blockquote>
      </li>
      <li
        className="carousel__indicator"
      >
        <span
          className="carousel__dot active"
        />
        <span
          className="carousel__dot"
        />
        <span
          className="carousel__dot"
        />
        <span
          className="carousel__dot"
        />
        <span
          className="carousel__dot"
        />
      </li>
    </ul>
  </div>
</div>
`;
